### PR TITLE
feat: write logical region to metric engine

### PR DESCRIPTION
**Pull request by waynexia**
I hereby agree to the terms of the [GreptimeDB CLA](https://gist.github.com/xtang/6378857777706e568c1949c7578592cc)

## What's changed and what's your intention?

Handle region put request in metric engine. Only put request to logical region is allowed for now.

This PR also changes the internal column `__metric` to `__table_id`. Instead of using metric name (or table name), table id is more suitable at this region engine layer.

## Checklist

- [ ]  I have written the necessary rustdoc comments.
- [ ]  I have added the necessary unit tests and integration tests.

## Refer to a related PR or issue link (optional)
- #2070

### CODE REVIEW

File: src/metric-engine/src/engine.rs [hunk 1]
```diff
@@ -687,14 +722,192 @@ impl MetricEngineInner {
...
+            .state
+            .read()
+            .await
+            .logical_regions
+            .get(&logical_region_id)
+            .with_context(|| LogicalRegionNotFoundSnafu {
+                region_id: logical_region_id,
+            })?;
+        let data_region_id = utils::to_data_region_id(physical_region_id);
+        let metadata_region_id = utils::to_metadata_region_id(physical_region_id);
```
Comment (fengjiachun) [inline src/metric-engine/src/engine.rs:766]: unused

File: src/metric-engine/src/engine.rs [hunk 2]
```diff
@@ -687,14 +722,192 @@ impl MetricEngineInner {
...
+                region_id: logical_region_id,
+            }
+            .fail();
+        }
+
+        // check if the columns exist
+        for col in &request.rows.schema {
+            if self
+                .metadata_region
+                .column_semantic_type(metadata_region_id, logical_region_id, &col.column_name)
```
Comment (fengjiachun) [inline src/metric-engine/src/engine.rs:806]: Can we provide a batch method? It costs a lot to call async methods in a loop.

Comment (waynexia) [reply src/metric-engine/src/engine.rs:806]: It's hard to do batch point get on mito engine. I can batch puts if a similar pattern occurs in the future.

File: src/metric-engine/src/engine.rs [hunk 3]
```diff
@@ -687,14 +722,191 @@ impl MetricEngineInner {
...
+        rows.schema.push(PbColumnSchema {
+            column_name: DATA_SCHEMA_TSID_COLUMN_NAME.to_string(),
+            datatype: to_column_data_type(&ConcreteDataType::uint64_datatype())
+                .unwrap()
+                .into(),
+            semantic_type: SemanticType::Tag as _,
+        });
+
+        // fill internal columns
+        let mut random_state = ahash::RandomState::with_seeds(1, 2, 3, 4);
```
Comment (evenyag) [inline src/metric-engine/src/engine.rs:869]: Can we define the state as a constant?
```
let mut random_state = RANDOM_STATE.clone();
```

File: src/metric-engine/src/engine.rs [hunk 4]
```diff
@@ -687,14 +722,191 @@ impl MetricEngineInner {
...
+            name.hash(&mut hasher);
+            // The type is checked before. So only null is ignored.
+            if let Some(ValueData::StringValue(string)) = tag.value_data {
+                string.hash(&mut hasher);
+            }
+        }
+        let hash = hasher.finish();
+
+        // fill table name and tsid
+        row.values.push(ValueData::U32Value(table_id).into());
```
Comment (evenyag) [inline src/metric-engine/src/engine.rs:896]:
```suggestion
        // fill table id and tsid
        row.values.push(ValueData::U32Value(table_id).into());
```

**Review by evenyag — APPROVED** [review]

**Review by fengjiachun — APPROVED** [review]
LGTM

**Pull request merged**